Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py

```diff
@@ -239,116 +32,51 @@ def translate_score(score: int, threshold: int) -> int:
-    cpe_regex = re.compile(
-        r'cpe:2\.3:(?P<Part>[aoh*-]):(?P<Vendor>[a-zA-Z_-]+):'
-        r'(?P<Product>[A-Za-z0-9_-]+):(?P<Version>[0-9.]+):(?P<Update>[a-zA-Z0-9*]+):'
-        r'(?P<Edition>[a-zA-Z0-9*]+):(?P<Language>[a-zA-Z0-9*]+):'
-        r'(?P<swedition>[a-zA-Z0-9*]+):(?P<targetsw>[a-zA-Z0-9*]+):'
-        r'(?P<targethw>[a-zA-Z0-9*]+):(?P<Other>[a-zA-Z0-9*]+)'
-    )
-    for cpe in cpes:
-        try:
-            match = cpe_regex.match(cpe)
-            if match:
-                tmp_dict = match.groupdict()
-                tmp_dict['Part'] = parts[tmp_dict.get('Part', '-')]
-                tmp_dict['Software Edition'] = tmp_dict.pop('swedition')
-                tmp_dict['Target Software'] = tmp_dict.pop('targetsw')
-                tmp_dict['Target Hardware'] = tmp_dict.pop('targethw')
-                dicts.append(tmp_dict)
-        except:
-            continue
-    return dicts
-
-
 def determine_hash(hash_value: str) -> str:
     """Determine hash type by length."""
     hash_length = len(hash_value)
     if hash_length == 128:
-        return "SHA512"
+        return 'SHA512'
     elif hash_length == 64:
-        return "SHA256"
+        return 'SHA256'
     elif hash_length == 40:
-        return "SHA1"
+        return 'SHA1'
     elif hash_length == 32:
-        return "MD5"
+        return 'MD5'
     elif hash_length == 8:
-        return "CRC32"
+        return 'CRC32'
     else:
-        return "CTPH"
-
-
-def level_to_criticality(level: int) -> str:
-    """Translate level integer to Criticality string."""
-    if level >= 4:
-        return "Very Malicious"
-    elif level >= 3:
-        return "Malicious"
-    elif level >= 2:
-        return "Suspicious"
-    elif level >= 1:
-        return "Informational"
-    return "Unknown"
-
-
-def rf_type_to_xsoar_type(entity_type: str) -> str:
-    if entity_type == "IpAddress":
-        return "ip"
-    elif entity_type == "Hash":
-        return "file"
-    elif entity_type == "URL":
-        return "url"
-    elif entity_type == "CyberVulnerability":
-        return "cve"
-    elif entity_type == "InternetDomainName":
-        return "domain"
-    raise DemistoException(
-        f"Unknown Recorded Future " f"entity type: {entity_type}"
-    )
-
-
-def prettify_time(time_string: str) -> str:
-    """Fix timestamps to a better format."""
-    if time_string:
-        parsed = datetime.strptime(time_string, "%Y-%m-%dT%H:%M:%S.%fZ")
-        return datetime.strftime(parsed, "%Y-%m-%d %H:%M:%S")
-    else:
-        return "N/A"
+        return 'CTPH'
 
 
 def create_indicator(
-    entity: str,
-    entity_type: str,
-    score: int,
-    description: str,
-    location: Dict[str, Any] = {},
+        entity: str,
+        entity_type: str,
+        score: int,
+        description: str = '',
+        location: Dict[str, Any] = None,
 ) -> Common.Indicator:
     """Create an Indicator object."""
     demisto_params = demisto.params()
+
+    if location is None:
```

Comment:
Why not just set the default value for location to dict() or {}?

Reply:
we don't want to have default values be mutable

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/